### PR TITLE
Use ALPN in rustls

### PR DESCRIPTION
**Pull request by jsha**

### CODE REVIEW

File: lib/vtls/rustls.c [hunk 1]
```diff
@@ -321,13 +321,24 @@ cr_init_backend(struct Curl_easy *data, struct connectdata *conn,
...
+    { (const uint8_t *)ALPN_H2, ALPN_H2_LENGTH },
+  };
 
   backend->tlsbuf = calloc(TLSBUF_SIZE, 1);
   if(!backend->tlsbuf) {
     return CURLE_OUT_OF_MEMORY;
   }
 
   config_builder = rustls_client_config_builder_new();
+#ifdef USE_NGHTTP2
```
Comment (kevinburkemeter) [inline lib/vtls/rustls.c:335]: This should be USE_HTTP2 now right?

**Comment by bagder** [comment]
The [travis build](https://travis-ci.org/github/curl/curl/jobs/768894278#L1591) is unhappy:
~~~
vtls/rustls.c: In function ‘cr_init_backend’:
vtls/rustls.c:337:3: error: implicit declaration of function ‘rustls_client_config_builder_set_protocols’; did you mean ‘rustls_server_config_builder_set_protocols’? [-Wimplicit-function-declaration]
   rustls_client_config_builder_set_protocols(config_builder, alpn, 2);
   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   rustls_server_config_builder_set_protocols
vtls/rustls.c:337:3: error: nested extern declaration of ‘rustls_client_config_builder_set_protocols’ [-Werror=nested-externs]
vtls/rustls.c: In function ‘cr_set_negotiated_alpn’:
vtls/rustls.c:393:3: error: implicit declaration of function ‘rustls_client_session_get_alpn_protocol’; did you mean ‘rustls_client_session_read_tls’? [-Wimplicit-function-declaration]
   rustls_client_session_get_alpn_protocol(session, &protocol, &len);
   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   rustls_client_session_read_tls
vtls/rustls.c:393:3: error: nested extern declaration of ‘rustls_client_session_get_alpn_protocol’ [-Werror=nested-externs]
cc1: all warnings being treated as errors

**Comment by bagder** [comment]
Ah, I guess that is because of the waiting PRs mentioned here already...

**Comment by jsha** [comment]
Oops, sorry about that. The crustls PR was merged, but I needed to tag a new release (0.5.0). I've done that now and updated .travis.yml and docs/RUSTLS.md to match.

**Review by bagder — APPROVED** [review]

**Comment by bagder** [comment]
Thanks!

**Pull request closed without merging**